src/core/component.ts

```diff
@@ -0,0 +1,3 @@
+export interface Component {
+    Process(input: any): any;
+}
```

Comment:
There are some things that simply aren't interchangeable, or don't need to share an interface. For example, people have stopped considering individual characters to be number-like, despite how C sees the world.

What does having this interface enable us to do that would be difficult without having the interface?